"use strict";(self.webpackChunk_0_l_documentation=self.webpackChunk_0_l_documentation||[]).push([[433],{3905:(e,t,o)=>{o.d(t,{Zo:()=>p,kt:()=>m});var n=o(7294);function r(e,t,o){return t in e?Object.defineProperty(e,t,{value:o,enumerable:!0,configurable:!0,writable:!0}):e[t]=o,e}function a(e,t){var o=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),o.push.apply(o,n)}return o}function s(e){for(var t=1;t<arguments.length;t++){var o=null!=arguments[t]?arguments[t]:{};t%2?a(Object(o),!0).forEach((function(t){r(e,t,o[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(o)):a(Object(o)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(o,t))}))}return e}function i(e,t){if(null==e)return{};var o,n,r=function(e,t){if(null==e)return{};var o,n,r={},a=Object.keys(e);for(n=0;n<a.length;n++)o=a[n],t.indexOf(o)>=0||(r[o]=e[o]);return r}(e,t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(n=0;n<a.length;n++)o=a[n],t.indexOf(o)>=0||Object.prototype.propertyIsEnumerable.call(e,o)&&(r[o]=e[o])}return r}var l=n.createContext({}),d=function(e){var t=n.useContext(l),o=t;return e&&(o="function"==typeof e?e(t):s(s({},t),e)),o},p=function(e){var t=d(e.components);return n.createElement(l.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},c=n.forwardRef((function(e,t){var o=e.components,r=e.mdxType,a=e.originalType,l=e.parentName,p=i(e,["components","mdxType","originalType","parentName"]),c=d(o),m=r,h=c["".concat(l,".").concat(m)]||c[m]||u[m]||a;return o?n.createElement(h,s(s({ref:t},p),{},{components:o})):n.createElement(h,s({ref:t},p))}));function m(e,t){var o=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var a=o.length,s=new Array(a);s[0]=c;var i={};for(var l in t)hasOwnProperty.call(t,l)&&(i[l]=t[l]);i.originalType=e,i.mdxType="string"==typeof e?e:r,s[1]=i;for(var d=2;d<a;d++)s[d]=o[d];return n.createElement.apply(null,s)}return n.createElement.apply(null,o)}c.displayName="MDXCreateElement"},6285:(e,t,o)=>{o.r(t),o.d(t,{assets:()=>l,contentTitle:()=>s,default:()=>u,frontMatter:()=>a,metadata:()=>i,toc:()=>d});var n=o(7462),r=(o(7294),o(3905));const a={},s="Documentation on Past Issues and Possible solutions",i={unversionedId:"node-ops/validator/guides/troubleshooting_onboarding",id:"node-ops/validator/guides/troubleshooting_onboarding",title:"Documentation on Past Issues and Possible solutions",description:"Issues",source:"@site/docs/node-ops/validator/guides/troubleshooting_onboarding.md",sourceDirName:"node-ops/validator/guides",slug:"/node-ops/validator/guides/troubleshooting_onboarding",permalink:"/0L-documentation/docs/node-ops/validator/guides/troubleshooting_onboarding",draft:!1,editUrl:"https://github.com/nControl88/0l-documentation-source/tree/main/docs/node-ops/validator/guides/troubleshooting_onboarding.md",tags:[],version:"current",frontMatter:{},sidebar:"tutorialSidebar",previous:{title:"Recreate all configs",permalink:"/0L-documentation/docs/node-ops/validator/guides/resetting_val_configs"},next:{title:"Update Validator to Operator mapping",permalink:"/0L-documentation/docs/node-ops/validator/guides/update_validator_to_operator_mapping"}},l={},d=[{value:"Issues",id:"issues",level:2},{value:'<a id="issue-1"></a> Issue &quot;Too many open files&quot;',id:"-issue-too-many-open-files",level:2},{value:"Validator logs show &quot;Too many open files&quot; or &quot;File descriptor limit exceeded&quot; error and tower app is stopped",id:"validator-logs-show-too-many-open-files-or-file-descriptor-limit-exceeded-error-and-tower-app-is-stopped",level:3},{value:'<a id="issue-2"></a> Issue: Epochs are not consecutive',id:"-issue-epochs-are-not-consecutive",level:2},{value:"Tower app starts but throws error on transaction: Failed to get state proof with error...Epochs are not consecutive",id:"tower-app-starts-but-throws-error-on-transaction-failed-to-get-state-proof-with-errorepochs-are-not-consecutive",level:3},{value:'<a id="issue-3"></a> Issue: DB should open... libradb/LOCK: Permission denied',id:"-issue-db-should-open-libradblock-permission-denied",level:2},{value:'<a id="issue-4"></a> Issue: Tower App start: EOF error',id:"-issue-tower-app-start-eof-error",level:2},{value:'<a id="issue-5"></a> Issue: Received response but no remote state found',id:"-issue-received-response-but-no-remote-state-found",level:2},{value:'<a id="issue-6"></a> Issue: NoAvailablePeers after the fullnode (diem-node) has been started.',id:"-issue-noavailablepeers-after-the-fullnode-diem-node-has-been-started",level:2},{value:'<a id="issue-7"></a> Issue: When trying to start the web monitor, Connection Failed: Connection refused (os error 111)',id:"-issue-when-trying-to-start-the-web-monitor-connection-failed-connection-refused-os-error-111",level:2}],p={toc:d};function u(e){let{components:t,...o}=e;return(0,r.kt)("wrapper",(0,n.Z)({},p,o,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"documentation-on-past-issues-and-possible-solutions"},"Documentation on Past Issues and Possible solutions"),(0,r.kt)("h2",{id:"issues"},"Issues"),(0,r.kt)("ol",null,(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-1"},'Validator logs show "Too many open files" or "File descriptor limit exceeded" error and tower app is stopped')),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-2"},"Tower start: Epochs not consecutive")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-3"},"DB should open")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-4"},"Tower start: EOF error")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-5"},"When trying to start the tower app, a response was received from an upstream_node but not a remote tower state")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-6"},"When a fullnode (diem-node) has started, in the logs it states that a response was received from an upstream_node but not a remote tower state.")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("a",{parentName:"li",href:"#issue-7"},"When trying to start the web monitor, Connection Failed: Connection refused (os error 111)"))),(0,r.kt)("h2",{id:"-issue-too-many-open-files"},(0,r.kt)("a",{id:"issue-1"}),' Issue "Too many open files"'),(0,r.kt)("h3",{id:"validator-logs-show-too-many-open-files-or-file-descriptor-limit-exceeded-error-and-tower-app-is-stopped"},'Validator logs show "Too many open files" or "File descriptor limit exceeded" error and tower app is stopped'),(0,r.kt)("p",null,'The validator is not voting or syncing and the file descriptor limit exceeded error is shown in the node logs. Tower app is stuck at a proof unable to submit the transaction with ".....".'),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Temporary Solution:")),(0,r.kt)("p",null,"This is an issue with the file descriptor limit being exceeded."),(0,r.kt)("ol",null,(0,r.kt)("li",{parentName:"ol"},"Go to the ",(0,r.kt)("strong",{parentName:"li"},"diem-node.service")," template you are using for starting the node."),(0,r.kt)("li",{parentName:"ol"},"Update ",(0,r.kt)("strong",{parentName:"li"},"LimitNOFILE=200000")),(0,r.kt)("li",{parentName:"ol"},"Restart node: ",(0,r.kt)("strong",{parentName:"li"},"make daemon")),(0,r.kt)("li",{parentName:"ol"},"Restart tower app: ",(0,r.kt)("strong",{parentName:"li"},"tower -o start"))),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"Note: This is until we figure out a better way not to reach the limit.")),(0,r.kt)("h2",{id:"-issue-epochs-are-not-consecutive"},(0,r.kt)("a",{id:"issue-2"})," Issue: Epochs are not consecutive"),(0,r.kt)("h3",{id:"tower-app-starts-but-throws-error-on-transaction-failed-to-get-state-proof-with-errorepochs-are-not-consecutive"},"Tower app starts but throws error on transaction: Failed to get state proof with error...Epochs are not consecutive"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},'Backlog: resubmitting missing blocks.\nThe application panicked (crashed).\nMessage:  called `Result::unwrap()` on an `Err` value: Failed to get state proof with error: JsonRpcError { code: -32000, message: "Server error: Epochs are not consecutive.", data: None }\nLocation: tower/src/submit_tx.rs:...\n\nBacktrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.\nRun with RUST_BACKTRACE=full to include source snippets.\n\n')),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Solution:")),(0,r.kt)("p",null,"It is possible that waypoint that tower app is using might be wrong. You can check the waypoint by looking at the logs on tower start."),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},' Waypoint: No waypoint parsed from command line args. Searching for waypoint in key_store.json\n[tower/src/config.rs:...] &value = "xxxxx"\n')),(0,r.kt)("p",null,'If it is not the right one. This needs to be updated in key_store.json under the key: "oper/waypoint". Update the value for this key with the waypoint your node synced from.'),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Fetching Waypoint:")),(0,r.kt)("ol",null,(0,r.kt)("li",{parentName:"ol"},"Use the waypoint value in restore_waypoint."),(0,r.kt)("li",{parentName:"ol"},"If the file is not there, this needs to be fetched from an existing node.")),(0,r.kt)("h2",{id:"-issue-db-should-open-libradblock-permission-denied"},(0,r.kt)("a",{id:"issue-3"})," Issue: DB should open... libradb/LOCK: Permission denied"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"2021-04-08T18:20:16.758813Z [main] ERROR common/crash-handler/src/lib.rs:38 details = '''panicked at 'DB should open.: IO error: while open a file for lock: /home/ubuntu/.0L/db/libradb/LOCK: Permission denied', diem-node/src/lib.rs:285:10'''\n")),(0,r.kt)("p",null,"You may be running ",(0,r.kt)("inlineCode",{parentName:"p"},"diem-node")," in a separate process. In that case there may be concurrent writes happening."),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Solution:")),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"Stop all diem-node instances. ",(0,r.kt)("inlineCode",{parentName:"li"},"killall diem-node")," or if you used makefile to start systemd: ",(0,r.kt)("inlineCode",{parentName:"li"},"make stop")),(0,r.kt)("li",{parentName:"ul"},"Restart diem-node")),(0,r.kt)("h2",{id:"-issue-tower-app-start-eof-error"},(0,r.kt)("a",{id:"issue-4"})," Issue: Tower App start: EOF error"),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Problem:")),(0,r.kt)("p",null,"Validator node ran out of space and tower app created empty block","_",".json file. When starting tower app below error is observed:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"Message: Error EOF while parsing a value\nLocation: tower/src/block.rs:....\n")),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Solution:")),(0,r.kt)("p",null,"Check if the last block proof created is empty. If so, remove the file and start the tower app again. This is after the node is caught up on the network."),(0,r.kt)("h2",{id:"-issue-received-response-but-no-remote-state-found"},(0,r.kt)("a",{id:"issue-5"})," Issue: Received response but no remote state found"),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Problem:")),(0,r.kt)("p",null,"When trying to start the tower app, a response was received from an upstream_node but not a remote tower state."),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"Message:  called `Result::unwrap()` on an `Err` value: Info: Received response but no remote state found. Exiting.\nLocation: ol/tower/src/backlog.rs:27\n")),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Solution:")),(0,r.kt)("p",null,"There could be an issue with the address used for the validator, or an issue with mining the genesis block.\nIf you have another address that has already been onboarded and being used for mining normally, you can try the following:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"change your ",(0,r.kt)("inlineCode",{parentName:"li"},"acount")," and ",(0,r.kt)("inlineCode",{parentName:"li"},"auth_key")," set in ",(0,r.kt)("inlineCode",{parentName:"li"},"~/.0L/0L.toml")," under ",(0,r.kt)("inlineCode",{parentName:"li"},"[profile]")," with details of your existing address."),(0,r.kt)("li",{parentName:"ul"},"start tower again with ",(0,r.kt)("inlineCode",{parentName:"li"},"tower start"),", ",(0,r.kt)("inlineCode",{parentName:"li"},"tower -u <upstream-ip-address> start")," or with whatever flags you were trying to start tower with."),(0,r.kt)("li",{parentName:"ul"},"you should now see ",(0,r.kt)("inlineCode",{parentName:"li"},"Mining VDF Proof # 1")," or similar in the logs."),(0,r.kt)("li",{parentName:"ul"},"If this works, you should use this address as the validator. To do this, you will need to follow step 2.2 onwards again from ",(0,r.kt)("a",{parentName:"li",href:"../../node-ops/validators/validator_onboarding_hard_mode.md#2-generate-account-keys"},"the hard onboarding guide"),", making sure to use the mnemonic of the working address when using ",(0,r.kt)("inlineCode",{parentName:"li"},"onboard val"))),(0,r.kt)("p",null,"Alternatively, you can try a brand new address:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"make sure it is onboarded to the network already (not validator yet). If you are able to mine using Carpe on another device, it's highly likely this address will work."),(0,r.kt)("li",{parentName:"ul"},"change your ",(0,r.kt)("inlineCode",{parentName:"li"},"acount")," and ",(0,r.kt)("inlineCode",{parentName:"li"},"auth_key")," set in ",(0,r.kt)("inlineCode",{parentName:"li"},"~/.0L/0L.toml")," under ",(0,r.kt)("inlineCode",{parentName:"li"},"[profile]")," with details of your new address."),(0,r.kt)("li",{parentName:"ul"},"start tower again with ",(0,r.kt)("inlineCode",{parentName:"li"},"tower start"),", ",(0,r.kt)("inlineCode",{parentName:"li"},"tower -u <upstream-ip-address> start")," or with whatever flags you were trying to start tower with."),(0,r.kt)("li",{parentName:"ul"},"you should now see ",(0,r.kt)("inlineCode",{parentName:"li"},"Mining VDF Proof # 1")," or similar in the logs."),(0,r.kt)("li",{parentName:"ul"},"If this works, you should use this address as the validator. To do this, you will need to follow step 2.2 onwards again from ",(0,r.kt)("a",{parentName:"li",href:"../../node-ops/validators/validator_onboarding_hard_mode.md#2-generate-account-keys"},"the hard onboarding guide"),", making sure to use the mnemonic of the working address when using ",(0,r.kt)("inlineCode",{parentName:"li"},"onboard val"))),(0,r.kt)("p",null,"If this doesn't work, it can also be caused by ",(0,r.kt)("a",{parentName:"p",href:"#issue-7"},"When trying to start the web monitor, Connection Failed: Connection refused (os error 111)")),(0,r.kt)("h2",{id:"-issue-noavailablepeers-after-the-fullnode-diem-node-has-been-started"},(0,r.kt)("a",{id:"issue-6"})," Issue: NoAvailablePeers after the fullnode (diem-node) has been started."),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Problem:")),(0,r.kt)("p",null,"When a fullnode (diem-node) has started, in the logs it states that a response was received from an upstream_node but not a remote tower state."),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},'2022-02-05T11:55:46.052036Z [state-sync] ERROR state-sync/src/coordinator.rs:219 {"error":"NoAvailablePeers(\\"No peers to send chunk request to!\\")","event":"fail","name":"progress_check"}\n2022-02-05T11:55:46.551657Z [state-sync] WARN state-sync/src/coordinator.rs:1514 {"name":"timeout","version":23877219}\n2022-02-05T11:55:46.551690Z [state-sync] WARN state-sync/src/coordinator.rs:1567 {"event":"missing_peers","name":"send_chunk_request"}\n2022-02-05T11:55:46.551818Z [state-sync] ERROR state-sync/src/coordinator.rs:1547 {"error":"NoAvailablePeers(\\"No peers to send chunk request to!\\")","event":"send_chunk_request_fail","local_epoch":102,"name":"timeout","version":23877219}\n2022-02-05T11:55:46.551895Z [state-sync] ERROR state-sync/src/coordinator.rs:219 {"error":"NoAvailablePeers(\\"No peers to send chunk request to!\\")","event":"fail","name":"progress_check"}\n')),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Solution:")),(0,r.kt)("p",null,"All you need to do is: ",(0,r.kt)("inlineCode",{parentName:"p"},"ol restore")),(0,r.kt)("p",null,"The issue occured perhaps due to configs not fully updated after changing the address or mnemonic associated with the fullnode. ",(0,r.kt)("inlineCode",{parentName:"p"},"ol restore")," will set the waypoint and update the ",(0,r.kt)("inlineCode",{parentName:"p"},"key_Store.json")),(0,r.kt)("h2",{id:"-issue-when-trying-to-start-the-web-monitor-connection-failed-connection-refused-os-error-111"},(0,r.kt)("a",{id:"issue-7"})," Issue: When trying to start the web monitor, Connection Failed: Connection refused (os error 111)"),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Problem:")),(0,r.kt)("p",null,"When starting the web_monitor, you may get ",(0,r.kt)("inlineCode",{parentName:"p"},"Connection Failed: Connection refused (os error 111)"),"."),(0,r.kt)("p",null,"You may also either have issue 5. ",(0,r.kt)("a",{parentName:"p",href:"#issue-5"},"When trying to start the tower app, a response was received from an upstream_node but not a remote tower state"),", the solution here could fix that too."),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},'can make client but could not get metadata Error { inner: Inner { kind: Request, source: Some(ConnectionFailed("Connection refused (os error 111)")), json_rpc_error: None } }\n\nCaused by:\n    Connection Failed: Connection refused (os error 111)\nERROR: could not create client connection, message: Cannot connect to any JSON RPC peers in the list of upstream_nodes in 0L.toml\n')),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Solution:")),(0,r.kt)("p",null,"There may be a bad IP address, ",(0,r.kt)("inlineCode",{parentName:"p"},"upstream_nodes"),", being used in ",(0,r.kt)("inlineCode",{parentName:"p"},"~/.0L/0L.toml"),". Try changing the ip address there to one from ",(0,r.kt)("a",{parentName:"p",href:"https://github.com/OLSF/carpe/blob/main/seed_peers/fullnode_seed_playlist.json"},"here"),".\n::"))}u.isMDXComponent=!0}}]);